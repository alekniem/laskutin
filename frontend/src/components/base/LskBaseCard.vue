<template>
    <v-card v-bind="$attrs">
        <slot name="header" />
        <v-divider />
        <slot />
    </v-card>
</template>

<script>
export default {
    name: "LskBaseCard",
};
</script>
