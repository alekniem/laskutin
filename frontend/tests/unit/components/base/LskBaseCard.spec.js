import LskBaseCard from "@/components/base/LskBaseCard";
import { createWrapper } from "UT/lsk_test_helper";

describe("LskBaseCard", () => {
    it("should match snapshot", () => {
        const wrapper = createWrapper(LskBaseCard);
        expect(wrapper.html()).toMatchSnapshot("no content in any slot");
    });

    it("should match snapshot", () => {
        const wrapper = createWrapper(LskBaseCard, {
            slots: {
                default: "<div>DEFAULT</div>",
            },
        });
        expect(wrapper.html()).toMatchSnapshot("content in default slot");
    });

    it("should match snapshot", () => {
        const wrapper = createWrapper(LskBaseCard, {
            slots: {
                header: "<div>HEADER</div>",
            },
        });
        expect(wrapper.html()).toMatchSnapshot("content in header slot");
    });

    it("should match snapshot", () => {
        const wrapper = createWrapper(LskBaseCard, {
            slots: {
                default: "<div>DEFAULT</div>",
                header: "<div>HEADER</div>",
            },
        });
        expect(wrapper.html()).toMatchSnapshot("content in default slot and in header slot");
    });
});
