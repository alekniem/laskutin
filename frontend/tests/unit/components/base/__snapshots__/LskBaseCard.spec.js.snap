// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LskBaseCard should match snapshot: content in default slot 1`] = `
<div class="v-card v-sheet theme--light">
  <hr role="separator"
      aria-orientation="horizontal"
      class="v-divider theme--light"
  >
  <div>
    DEFAULT
  </div>
</div>
`;

exports[`LskBaseCard should match snapshot: content in default slot and in header slot 1`] = `
<div class="v-card v-sheet theme--light">
  <div>
    HEADER
  </div>
  <hr role="separator"
      aria-orientation="horizontal"
      class="v-divider theme--light"
  >
  <div>
    DEFAULT
  </div>
</div>
`;

exports[`LskBaseCard should match snapshot: content in header slot 1`] = `
<div class="v-card v-sheet theme--light">
  <div>
    HEADER
  </div>
  <hr role="separator"
      aria-orientation="horizontal"
      class="v-divider theme--light"
  >
</div>
`;

exports[`LskBaseCard should match snapshot: no content in any slot 1`] = `
<div class="v-card v-sheet theme--light">
  <hr role="separator"
      aria-orientation="horizontal"
      class="v-divider theme--light"
  >
</div>
`;
